lab-davidw/test/signup-test.js

```diff
@@ -0,0 +1,47 @@
+'use strict';
+
+const expect = require('chai').expect;
+
+describe('testing signup', function(){
+  beforeEach(done => {
+    angular.mock.module('cfgram');
+    angular.mock.inject(($rootScope, $SignupController, $httpBackend) => {
+      this.$rootScope = $rootScope;
+      this.$httpBackend = $httpBackend;
+      this.signupCtrl = $SignupController('signup');
+    });
+  });
+
+  afterEach(done => {
+    this.$httpBackend.verifyNoOutstandingExpectation();
+    this.$httpBackend.verifyNoOutstandingRequest();
+  });
+
+  afterEach(done => this.$rootScope.$apply());
+
+  it('should send user data to the server', () => {
+    let expectUrl = 'http://localhost:3000/api/auth/signup';
+    let expectBody = {
+      username: 'testUser',
+      email: 'testuser@test.com',
+      password: 'pass123',
```

Comment:
You'll also want to test out your edit controller now. By first creating a mock gallery and then passing that context to your update function